lib/datadog/kit/appsec/events.rb

```diff
@@ -14,56 +14,101 @@ module Events
         #
         # This method is experimental and may change in the future.
         #
-        # @param trace [TraceOperation] Trace to attach data to.
+        # @param trace [TraceOperation] Trace to attach data to. Defaults to
+        #   active trace.
+        # @param span [SpanOperation] Span to attach data to. Defaults to
+        #   active span on trace. Note that this should be a service entry span.
+        #   When AppSec is enabled, the expected span and trace are automatically
+        #   used as defaults.
         # @param user [Hash<Symbol, String>] User information to pass to
         #   Datadog::Kit::Identity.set_user. Must contain at least :id as key.
         # @param others [Hash<String || Symbol, String>] Additional free-form
         #   event information to attach to the trace.
-        def self.track_login_success(trace, user:, **others)
-          track(LOGIN_SUCCESS_EVENT, trace, **others)
+        def self.track_login_success(trace = nil, span = nil, user:, **others)
+          if (appsec_scope = Datadog::AppSec.active_scope)
+            trace = appsec_scope.trace
+            span = appsec_scope.service_entry_span
+          end
+
+          trace ||= Datadog::Tracing.active_trace
+          span ||= trace.active_span || Datadog::Tracing.active_span
+
+          raise ArgumentError, "span #{span.span_id} does not belong to trace #{trace.id}" if trace.id != span.trace_id
```

Comment:
Should we raise or log an message letting the user that they have done something odd?

Reply:
As with the active scope (prev. active context) sanity checks, this should not happen, ever. If we swallow it we'll never be aware.